package/MDAnalysis/analysis/MSD.py

```diff
@@ -0,0 +1,468 @@
+            pos index, organized:
+            selection X frame X dict
+        this_restart : int
+            frame number of the current restart
+        """
+        # stop updating when there are no more frames to analyze
+        top_frame = this_restart + self.len_msd
+        calc_cutoff = self.tf
+        if self.max_data:
+            calc_cutoff = len(self.universe.trajectory) - 1
+
+        if top_frame+self.dt_restart > calc_cutoff:
+            # feed in zeros
+            for i in range(len(self.select_list)):
+                pos[i].append(np.array([]))
+                set_list[i].append(set())
+                dict_list[i].append(dict())
+            return
+        
+        for ts in self.universe.trajectory[top_frame:top_frame
+                                           + self.dt_restart]:
+            selections = self._select(ts.frame)
+
+            for i in range(len(selections)):
+                if selections[i] is not None:  # if there is data
+                    pos[i].append(selections[i].positions)
+
+                    temp_list = selections[i].atoms.ids
+                    # store set of atom ids which satisfy selection
+                    set_list[i].append(set(temp_list))
+                    # link atom id to array index
+                    temp_dict = dict()
+                    for j in range(len(temp_list)):
+                        temp_dict[temp_list[j]] = j
+                    dict_list[i].append(temp_dict)
+                else:
+                    pos[i].append(np.array([]))
+                    set_list[i].append(set())
+                    dict_list[i].append(dict())
+
+    def _process_pos_data(self, pos, set_list, dict_list):
+        """Runs MSD calculation and returns data
+
+        Parameters:
+        -----------
+        pos : list
+            contains position information, organized:
+            selection X frame X atom X (x, y, z)
+        set_list : list
+            contains set of atoms which satisfy each selection at each
+            frame, organized:
+            selection X frame X set of atom ids
+        dict_list : list
+            contains dictionaries which map atom ids to corresponding
+            pos index, organized:
+            selection X frame X dict
+
+        Returns:
+        --------
+        msd : np.array(dtype=float32)
+            Contains msd data, organized:
+            selection X tau
+        n_samples : np.array(dtype=int)
+            Contains the number of samples corresponding to each msd
+            entry, organized:
+            selection X tau
+        """
+        n_sel = len(pos)
+
+        n_samples = np.zeros((n_sel, self.len_msd), dtype='int')
+        msd = np.zeros((n_sel, self.len_msd), dtype='float32')
+        
+        calc_cutoff = self.tf
+        if self.max_data:
+            calc_cutoff = len(self.universe.trajectory) - 1
+
+        # for each restart point
+        for j in range(self.t0, self.tf+1, self.dt_restart):
```

Comment:
So `dt_restart` is the steps between frames that are analyzed for the MSD. Then please call it `step` to be consistent with the default python/MDAnalysis terminology.